src/c/include/pn/procyon.h

```diff
@@ -24,14 +24,15 @@
 extern "C" {
 #endif  // __cplusplus
 
-typedef struct pn_value   pn_value_t;
-typedef struct pn_data    pn_data_t;
-typedef struct pn_string  pn_string_t;
-typedef struct pn_array   pn_array_t;
-typedef struct pn_map     pn_map_t;
-typedef struct pn_kv_pair pn_kv_pair_t;
-typedef struct pn_input   pn_input_t;
-typedef struct pn_output  pn_output_t;
+typedef struct pn_value     pn_value_t;
+typedef struct pn_data      pn_data_t;
+typedef struct pn_string    pn_string_t;
+typedef struct pn_string16  pn_string16_t;
```

Comment:
I would rather not have two different string types in the public interface of this library. It seems to be orthogonal to the actual purpose of this change (supporting UTF-16 in `pn_path_input()` / `pn_path_output()`).

On the other hand, it would probably be useful to have string constructors from `wchar_t*` / `std::wstring`.

Reply:
Yeah not sure where this should live... problem is it's needed in libsfz too.  Maybe it'd be better to move this to a separate library.

Problem with wchar_t is it's implementation-defined, it's UTF-32 on Linux despite not being useful there, there are still no standard encode functions in C, and the C++ ones are being revised.  So, it really only makes sense to implement UTF-16 for dealing with file paths.

Reply:
Could you use `char16_t` / `std::u16string` for Windows? libsfz can assume C++11 support.